src/pages/GroupPage.js

```diff
@@ -0,0 +1,187 @@
+import React, { useState, useRef, useEffect } from 'react';
+import {
+  Form,
+  Helmet,
+  withHistory,
+  formatMessage,
+  formatMessageWithValues,
+  coreConfirm,
+  clearConfirm,
+  journalize,
+} from '@openimis/fe-core';
+import { injectIntl } from 'react-intl';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+import { withTheme, withStyles } from '@material-ui/core/styles';
+import DeleteIcon from '@material-ui/icons/Delete';
+import { RIGHT_GROUP_UPDATE } from '../constants';
+import { fetchGroup, deleteGroup, updateGroup } from '../actions';
+import GroupHeadPanel from '../components/GroupHeadPanel';
+import IndividualTabPanel from '../components/IndividualTabPanel';
+import { ACTION_TYPE } from '../reducer';
+
+const styles = (theme) => ({
+  page: theme.page,
+});
+
+function GroupPage({
+  intl,
+  classes,
+  rights,
+  history,
+  groupUuid,
+  group,
+  fetchGroup,
+  deleteGroup,
+  updateGroup,
+  coreConfirm,
+  clearConfirm,
+  confirmed,
+  submittingMutation,
+  mutation,
+  journalize,
+}) {
+  const [editedGroup, setEditedGroup] = useState({});
+  const [confirmedAction, setConfirmedAction] = useState(() => null);
+  const prevSubmittingMutationRef = useRef();
+
+  useEffect(() => {
+    if (groupUuid) {
+      fetchGroup([`id: "${groupUuid}"`]);
+    }
+  }, [groupUuid]);
+
+  useEffect(() => {
+    if (confirmed) confirmedAction();
+    return () => confirmed && clearConfirm(null);
+  }, [confirmed]);
+
+  const back = () => history.goBack();
+
+  useEffect(() => {
```

Comment:
can you try to reduce inner if by using this? This should work in the same way, you can check. If no please use the current solution
```
useEffect(() => {
  if (prevSubmittingMutationRef.current && !submittingMutation) {
    journalize(mutation);
    mutation?.actionType === ACTION_TYPE.DELETE_GROUP && back();
  }
}, [submittingMutation]);
```

Reply:
breaks eslint